deploy/azure/infra/main.tf

```diff
@@ -80,35 +80,122 @@ resource "azurerm_data_factory_managed_private_endpoint" "blob_pe" {
   subresource_name   = "blob"
 }
 
+resource "null_resource" "approve_adf_blob_private_endpoint" {
+
+  triggers = {
+    always_run = timestamp()
+  }
+  provisioner "local-exec" {
+    #  interpreter = ["sh", "-Command"]
+    command = <<-EOT
+        az login --service-principal -u ${data.azurerm_client_config.current.client_id} -p ${var.azure_client_secret} --tenant ${data.azurerm_client_config.current.tenant_id}
```

Comment:
If we still want to go with this approach then probably worth create a script e.g. "approve_private_endpoints.sh" then call this because there is a lot of repeated code.

Reply:
Removed repeated code ,added single null resource for approving all ADF PE .